### PR TITLE
Modify method to allow option to override scopes.

### PR DESCRIPTION
**Pull request by martinstreicher**

### CODE REVIEW

**Comment by martinstreicher** [comment]
Specs pass.

**Review by aditya-kapoor — APPROVED** [review]
🚀

File: lib/nylas.rb [hunk 1]
```diff
@@ -128,12 +128,12 @@ def url_for_path(path)
...
       params = {
-        :client_id => @app_id,
-        :trial => options.fetch(:trial, false),
+        :client_id     => @app_id,
+        :trial         => options.fetch(:trial, false),
         :response_type => 'code',
-        :scope => 'email',
-        :login_hint => login_hint,
-        :redirect_uri => redirect_uri,
+        :scope         => options.fetch(:scope, 'email'),
```
Comment (kyletolle) [inline lib/nylas.rb:134]: This defaults to the generic `email` scope in the case that more specific email scopes (like `'email.modify` and `email.send`) are not provided? If no `:scope` option was passed in, and `email` is used, is that equivalent to what our code currently does?

File: spec/inbox_spec.rb [hunk 2]
```diff
@@ -98,6 +98,19 @@
       expect(params["state"]).to eq('empire state')
     end
 
+    it 'should pass scope if defined' do
```
Comment (kyletolle) [inline spec/inbox_spec.rb:101]: What about if the scope is not defined?

File: lib/nylas.rb (hunk 1 above)
Comment (martinstreicher) [reply lib/nylas.rb:134]: This code functions identically to the current version of the gem if no options are passed in. And yes, this code would keep our app the same if no options are passed in.

File: spec/inbox_spec.rb (hunk 2 above)
Comment (martinstreicher) [reply spec/inbox_spec.rb:101]: Good question. I can add a test.

**Pull request merged**